docs/drivers.md

```diff
@@ -3,26 +3,98 @@ Bit components are language agnostic. You can write components in any language y
 
 **Important** To import and manage components use Bit-CLI. This doc only explains how to use components in your code.
 
 In this docs we will describe all currently available drivers.
 
+### Drivers are simply devDependencies
+
+Bit drivers are only required when developing or building projects. They are there simply to create the virtual library. Once the virtual library is set accoring to your project's bit.json file, your application will natively use the components, just like libraries.
+
+The only thing you need to do is to remember to pack the `components` directory alongside your project.
+
+### Component devDependencies
+
+Bit manages all of it's components devDependencies in it's own `components` folder, so not to interfere with your project's environment.
+
+### Component Runtime Dependencies
+
+When importing new components that have runtime dependencies to your code, the driver will evaluate if your runtime dependencies are met with the requirements of the component you just imported. If there is a mismatch, or your project is missing a dependency to allow the component to run, Bit will ask you to add it to your list of runtime dependencies (Bit also checks for SemVer competibility of dependencies).
+
+### Components in Runtime
+
+All runtime dependency resulotion of Bit components is done via the virtual API that the language-specific driver creates. This means that there are no special steps you need to take when running applications that using Bit. Not even adding any runtime dependency.
+
+## How Bit uses it's Drivers
+
+When any of Bit's lifecycle commands are triggered, Bit will call the driver's `bind` hook so it will update the virtual library for the specific environment.
+
+These are Bit's lifecycle commands.:
+
+* create
+* commit
+* modify
+* export
+* import
+
+The way that Bit Knows which driver it needs to look for is using the component's `language` attribute from it's [bit.json](configuring-bit.mdl#bitjson) file. Once Bit figured out which driver it needs, it will simply try to `require` for the driver in the directory that the command ran in. So if the driver is installed as a devDependency, it will be able to find it in the project's folder tree, just like any other module.
+
 ## JavaScript
 
 Bit's first fully supported programming language is JavaScript.
 
 You can find the bit-js driver codebase [here](https://github.com/teambit/bit-js).
 
-### Installing bit-js
+### JavaScript Virtual Library
+
+All your project dependencies for your project are located in the `node_modules` directory. So if you are using NodeJS, you can `require` any of these dependencies. Also, any bundleing process uses the same method to pack them alongside your code, to run later on a web-browser.
```

Comment:
> All your project dependencies for your project are located in the `node_modules` directory. So if you are using NodeJS, you can `require` any of these dependencies.

If you're using NodeJS, all the external dependencies for your project are located in the `node_modules` directory.
To run this code in a web browser, you can use [Webpack](https://webpack.js.org/)/[browserify](http://browserify.org/), which are module bundlers.